tool/tsh/proxy.go

```diff
@@ -333,12 +334,9 @@ func onProxyCommandDB(cf *CLIConf) error {
 		return trace.Wrap(err)
 	}
 
-	if routeToDatabase.Protocol == defaults.ProtocolSnowflake && !cf.LocalProxyTunnel {
-		return trace.BadParameter("Snowflake proxy works only in the tunnel mode. Please add --tunnel flag to enable it")
-	}
-
-	if routeToDatabase.Protocol == defaults.ProtocolElasticsearch && !cf.LocalProxyTunnel {
-		return trace.BadParameter("Elasticsearch proxy works only in the tunnel mode. Please add --tunnel flag to enable it")
+	requiresLocalProxy := []string{defaults.ProtocolSnowflake, defaults.ProtocolElasticsearch}
```

Comment:
Any reason `LocalProxyTunnel` is required? In my testing, things are fine when curl sends the request with the client cert.

Reply:
I expect most clients to be db clients utilising JDBC/ODBC drivers. These drivers are rather annoying to configure properly with certificates (possible in theory, error prone in practice). For those, `--tunnel` should provide good UX. But it is true that this isn't the only possible use case, so forcing `--tunnel` is probably wrong. Thanks, I'll drop this check.